sqlite/src/expr.c

```diff
@@ -5550,7 +5550,7 @@ static char *print_mem(Mem *m){
       }
       key[2*m->n] = '\0';
 
-      return sqlite3_mprintf("x'%s'", key);
+      return sqlite3_mprintf("x'%q'", key);
```

Comment:
I don't think we can have quote characters inside a hex literal (not that this case is any worse than without the change)